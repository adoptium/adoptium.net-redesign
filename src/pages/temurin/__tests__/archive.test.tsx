import React from "react"
import { act, render, waitFor } from "@testing-library/react"
import { afterEach, describe, expect, it, vi } from "vitest"
import { axe } from "vitest-axe"
import Archive, { Head } from "../archive"
import AxiosInstance from "axios"
import MockAdapter from "axios-mock-adapter"

const mock = new MockAdapter(AxiosInstance)

afterEach(() => {
  vi.clearAllMocks()
})

describe("Temurin Archive page", () => {
  it("renders correctly", async () => {
    mock.onGet().reply(200, [], { pagecount: 0 })

    let container
    await act(async () => {
      const renderResult = render(<Archive />)
      container = renderResult.container
    })
    // eslint-disable-next-line
    const pageContent = container.querySelector("main")

    expect(pageContent).toMatchSnapshot()
  })

  it("head renders correctly", async () => {
    mock.onGet().reply(200, [], { pagecount: 0 })

    const { container } = render(<Head />)
    // eslint-disable-next-line
    const title = container.querySelector("title")
    expect(title).toHaveTextContent("Archive | Adoptium")
  })

  it("has no accessibility violations", async () => {
    mock.onGet().reply(200, [], { pagecount: 0 });
    const { container } = render(<Archive />);
    await waitFor(() => {
      expect(container.querySelector("table")).toBeInTheDocument();
    });
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
})
