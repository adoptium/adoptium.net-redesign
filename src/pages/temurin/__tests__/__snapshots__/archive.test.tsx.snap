// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Temurin Archive page > renders correctly 1`] = `
<main>
  <section
    class="py-5 text-center container"
  >
    <div
      class="row py-lg-5"
    >
      <div
        class="col-lg-10 col-md-8 mx-auto"
      >
        <h1
          class="fw-light"
        >
          Archive
        </h1>
        <div
          class="row align-items-center pt-2"
        >
          <div
            class="callout callout-default text-start"
          >
            Text
            <br />
            <br />
            <p
              class="text-warning"
            >
              Text
            </p>
          </div>
          <div
            class="btn-group"
          >
            <a
              class="btn btn-primary m-3"
              href="/temurin/releases"
            >
              Text
               
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 8c137 0 248 111 248 248S393 504 256 504 8 393 8 256 119 8 256 8zm-28.9 143.6l75.5 72.4H120c-13.3 0-24 10.7-24 24v16c0 13.3 10.7 24 24 24h182.6l-75.5 72.4c-9.7 9.3-9.9 24.8-.4 34.3l11 10.9c9.4 9.4 24.6 9.4 33.9 0L404.3 273c9.4-9.4 9.4-24.6 0-33.9L271.6 106.3c-9.4-9.4-24.6-9.4-33.9 0l-11 10.9c-9.5 9.6-9.3 25.1.4 34.4z"
                />
              </svg>
            </a>
            <a
              class="btn btn-secondary m-3"
              href="/temurin/nightly"
            >
              Text
               
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 8c137 0 248 111 248 248S393 504 256 504 8 393 8 256 119 8 256 8zm-28.9 143.6l75.5 72.4H120c-13.3 0-24 10.7-24 24v16c0 13.3 10.7 24 24 24h182.6l-75.5 72.4c-9.7 9.3-9.9 24.8-.4 34.3l11 10.9c9.4 9.4 24.6 9.4 33.9 0L404.3 273c9.4-9.4 9.4-24.6 0-33.9L271.6 106.3c-9.4-9.4-24.6-9.4-33.9 0l-11 10.9c-9.5 9.6-9.3 25.1.4 34.4z"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </div>
    <p
      class="text-center"
    >
      Text
    </p>
    <div
      class="input-group p-3 d-flex justify-content-center"
    >
      <label
        class="px-2 fw-bold"
        for="version"
      >
        Text
      </label>
      <select
        aria-label="version-filter"
        class="form-select form-select-sm"
        data-testid="version-filter"
        id="version-filter"
        style="max-width: 10em;"
      >
        <option
          value="1"
        >
          1 - LTS
        </option>
        <option
          value="2"
        >
          2
        </option>
      </select>
    </div>
    <div
      id="archive-list"
    >
      <table
        class="table table-borderless table-condensed archive-container"
        id="archive-table"
        style="border-spacing: 0 20px; border-collapse: separate;"
      >
        <tbody
          class="table-light"
        />
      </table>
      0
    </div>
  </section>
</main>
`;
